onnxruntime/core/framework/session_options.h

```diff
@@ -32,7 +32,13 @@ struct SessionOptions {
   // enable profiling for this session.
   bool enable_profiling = false;
 
-  // non empty filepath enables serialization of the transformed optimized model to the specified filepath.
+  // Non empty filepath enables serialization of the transformed optimized model to the specified filepath.
+  //
+  // Set session config value for ORT_SESSION_OPTIONS_CONFIG_SAVE_MODEL_FORMAT to 'ORT' or 'ONNX' to explicitly
+  // specify the format.
+  //
+  // If session config value is not set, it will be assumed to be ONNX
+  // unless the filepath ends in '.ort' (case insensitive).
   std::basic_string<ORTCHAR_T> optimized_model_filepath;
 
```

Comment:
Do we want to add a separate path for ORT model using session configs? Mix ort output with onnx optimized output may be a bit confusing for end users?

Reply:
I don't want to clutter the config with two fields for something not used during 'normal' ORT usage. Given the name of the value just says 'optimized' it doesn't seem overly confusing to me given the comment explaining what will be produced. 

---
In reply to: [481373928](https://github.com/microsoft/onnxruntime/pull/4995#discussion_r481373928) [](ancestors = 481373928)